### PR TITLE
feat: align MoE baselines and restore optional SOL floors

### PR DESCRIPTION
**Pull request by mental2008**
## Overview

This PR aligns the public fused MoE input/baseline contract and restores an optional dispatch-latency floor for roofline SOL calculations. It also adds a static validator so cached SOL values can be checked without importing operator modules or running GPU kernels.

The changes are intentionally limited to the public core. Internal reporting, production adapters, private hardware profiles, and unredacted data are not included here.

### 1. Fused MoE input and baseline consistency

The previous public `fused_moe/input.py` generated `topk_ids` with independent `torch.randint` calls. That allowed the same token to select the same expert more than once, which is not representative of normal top-k router output. Duplicate routes can make sorted/scatter-style production kernels behave differently from the gather-based reference and can therefore create false correctness failures.

The public metadata also still contained the older SGLang performance baseline, while the corresponding production path and the maintained measurements use AITER. The input contract and the baseline therefore described different execution assumptions.

### 2. SOL values below dispatch latency

Pure roofline latency is computed as:

```text
max(sum(W[dtype] / P_peak[dtype]), Q / B_peak)
```

For very small kernels, that theoretical value can be lower than the measured kernel-dispatch latency of a device. The public implementation had no way to express that device-specific lower bound, even when a hardware profile had a measured value available. This could produce unrealistic sub-dispatch SOL values and make cached SOL values disagree with production-latency checks.

### Fused MoE routing and public baselines

- Generate router logits over all experts and use `torch.topk` to select routes.
- Guarantee that every token selects `top_k` distinct experts.
- Normalize the selected weights and preserve the public `float32`/`int32` contract.
- Update all 23 public XPU-A fused MoE baseline entries to the recorded AITER measurements.
- Preserve the rest of each public shape metadata object; no private trace/model provenance is introduced.

These baseline numbers are migrated recorded measurements. This PR did **not** re-run the production benchmark on a GPU.

### Optional launch-overhead floor

- Add optional `launch_overhead_s` support to `RooflineHardware`.
- Accept a missing value or `null` as “no floor,” preserving current behavior.
- Reject booleans, strings, zero/negative values, NaN, and infinities.
- Apply the floor consistently to:
  - single-dtype roofline calculations;
  - hybrid/multi-dtype calculations;
  - integer/memory-only operator calculations in the CLI.
- Keep an empty workload (`W == 0` and `Q == 0`) at zero.
- Add `clamped_by_overhead` to `RooflineResult` and JSON output.
- Label clamped values in text output so a launch floor is not mistaken for a pure theoretical roofline bound.

No device-specific floor is hardcoded in the implementation. A floor is active only when the selected hardware profile explicitly provides one.

### Static SOL validation

Add `scripts/static_roofline_check.py`, which performs read-only checks without loading `input.py`/`reference.py` or executing kernels:

- recomputes cached `SOL_time_ms` from semantic W/Q fields and hardware YAML;
- uses the same optional launch-overhead semantics as the Python API and roofline CLI;
- reports missing/invalid hardware mappings and W/Q inconsistencies;
- compares `T_SOL / TProd` only when the production baseline belongs to the same hardware key, avoiding cross-device comparisons;
- supports operator filters, explicit latency hardware, JSON/text output, configurable tolerances, and `--warnings-as-errors`.

### Documentation

`docs/roofline_semantics.md` documents:

- the theoretical SOL formula;
- the distinction between empty work and memory-only work;
- the optional dispatch floor and `clamped_by_overhead` flag;
- cache review with `--no-write` before explicit recomputation;
- the requirement to validate timing-mode compatibility, especially before applying an eager-dispatch floor to CUDA Graph replay;
- the provenance limitation of the migrated fused MoE baseline.

## Compatibility and migration behavior

- Hardware profiles without `launch_overhead_s`, or with `launch_overhead_s: null`, retain the previous theoretical result.
- The new dataclass fields have defaults, so existing constructors remain compatible.
- Existing cached `SOL_time_ms` values are **not** rewritten automatically by this PR.
- Bottleneck classification, arithmetic intensity, and roofline throughput continue to describe the theoretical model; only the reported latency may be raised by the explicit dispatch floor.
- JSON output gains the additive `clamped_by_overhead` field.

## Validation

Local CPU validation on the PR commit:

```text
Full test suite:       547 passed, 8 skipped
Focused new contracts: 69 passed
Static public-data SOL check: 0 errors, 0 warnings
```

The focused coverage includes:

- unique MoE routes for `top_k` values 1, 2, and 4;
- normalized route weights and expected dtypes;
- coverage of all 23 fused MoE shapes and absence of private provenance fields;
- single-dtype, hybrid, memory-only, and empty-work SOL paths;
- missing/null/valid/invalid launch-overhead values;
- CLI/API/static-check consistency and no-write cache behavior;
- exact hardware matching for production-latency comparisons.

GitHub CodeQL checks and the CLA check are currently passing.

## Not covered by this PR

- No new GPU performance measurement was performed.
- No public hardware profile is assigned a new launch-overhead value here.
- No existing SOL cache is bulk-refreshed.
- Internal reporting, private production reproduction tooling, private hardware specifications, and Aone CI changes remain in the internal extension repository.

## Suggested review order

1. `data/fused_moe/input.py` and `data/fused_moe/metadata.json` for the input/baseline contract.
2. `src/atrex_bench/eval/roofline.py` for API semantics and backward compatibility.
3. `scripts/roofline.py` and `scripts/static_roofline_check.py` for CLI consistency and failure policy.
4. The three new test modules and `docs/roofline_semantics.md` for edge cases and documented limitations.

### CODE REVIEW

**Review by YuxiaoWang-520 — APPROVED** [review]
Reviewed the public MoE contract, baseline migration, roofline floor semantics, and static validation path. No blocking issues found.
